src/helpers/iconpicker.class.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+var $           = require('jquery');
+var eventablejs = require('eventablejs');
+var _           = require('../lodash.js');
+var Modal       = require('etudiant-mod-modal');
+var xhr         = require('etudiant-mod-xhr');
+var Slider      = require('./slider.class.js');
+var FileUploader = require('../extensions/file-uploader.js');
+
+var iconPickerHtml = '<div class="icon-picker"><div class="droppable st-block__upload-container">' + i18n.t('blocks:illustrated:placeholder:drop') + '</div></div>';
+
+function triggerChangeIllustratedPicture(block, pictureInformations){
+    block.blockRef.imageId = pictureInformations.id;
+    block.trigger('picture:change', pictureInformations);
+
+    block.modal.close();
+
+}
+
+function bindClickOnIcons(block) {
+    $.each(block.slider.$elem.find('img'), function(){
+        $(this).on('click', function(){
+            triggerChangeIllustratedPicture(block, {
+                src: $(this).attr('src'),
+                copyright: $(this).attr('alt'),
+                id: $(this).data('id')
+            });
+        });
+    });
+}
+
+function dropEvent(iconPicker){
+    $(iconPicker.modal.$elem.children('.droppable')[0]).off();
+    $(iconPicker.modal.$elem.children('.droppable')[0]).on('dragover dragenter', function(ev){
+        ev.preventDefault();
+        ev.stopPropagation();
+    });
+
+
+    $(iconPicker.modal.$elem.children('.droppable')[0]).on('drop', function(ev){
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        iconPicker.blockRef.loading();
+
+        var file = ev.originalEvent.dataTransfer.files[0];
+
+        if (/image/.test(file.type)) {
+
+            var fileUploader = new FileUploader(iconPicker.blockRef, iconPicker.blockRef.globalConfig.apiUrl + 'edt/media/upload');
+
+            var uploadedMedia;
+
+            var uploadPromise = fileUploader.upload(file)
+            .then(function(returnedData){
+                uploadedMedia = iconPicker.blockRef.globalConfig.apiUrl + 'edt/media/' + returnedData.idMedia;
+            });
+
+            uploadPromise.then(function(){
+                iconPicker.blockRef.ready();
+                iconPicker.modal.close();
+                xhr.get(uploadedMedia)
+
+                .then(function(imageData) {
+                   triggerChangeIllustratedPicture(iconPicker, {
+                        src: imageData.content.thumbnail,
+                        copyright: imageData.content.copyright,
+                        id: imageData.content.id
+                    });
+                });
+            });
+
+            uploadPromise.catch(function(err){
+                console.log(err);
+            });
```

Comment:
@arthur-balagne this promise chain is poorly structured - I can't accept it in its current state